app/services/assignment_service.py

```diff
@@ -14,6 +14,51 @@
 class AssignmentService:
     def __init__(self, session: Session):
         self.session = session
+        
+    async def create_assignment(
+        self,
+        id: int,
+        name: str,
+        directory_path: str,
+        available_date: datetime,
+        due_date: datetime
+    ) -> AssignmentModel:
+
+        from app.services import GiteaService, FileOperation, FileOperationType, CourseService
+
+        gitea_service = GiteaService()
+        course_service = CourseService(self.session)
+
+        assignment = AssignmentModel(
+            id=id,
+            name=name,
+            directory_path=directory_path,
+            available_date=available_date,
+            due_date=due_date
+        )
+
+        self.session.add(assignment)
+        self.session.commit()
+
+        assignment_folder = f"{ name }_{ id }-dist"
+        master_notebook = f"{ name }-prof.ipynb"
+        master_repository_name = await course_service.get_master_repository_name()
+        owner = await course_service.get_instructor_gitea_organization_name()
+
+        files_to_modify = [
+            FileOperation(content="", path=f"{ assignment_folder }/{ master_notebook }", operation=FileOperationType.CREATE),
+            FileOperation(content=f"*grades.csv \n { master_notebook }", path=f"{ assignment_folder }/.gitignore", operation=FileOperationType.CREATE),
+        ]
+
+        await gitea_service.modify_repository_files(
+            name=master_repository_name,
+            owner=owner,
+            branch_name="master",
+            commit_message=f"Create assignment { name }",
```

Comment:
probably just "Initialize assignment" is fine. You should add a method to CourseService similar to get_master_repository_name for `get_main_branch_name`. should call a static method, which can literally just return "master"

Reply:
done